health_monitor/src/health_monitor.cpp

```diff
@@ -139,8 +153,223 @@ namespace health_monitor
+        auto crit_driver_status_car = driver_manager_.are_critical_drivers_operational_car(time_now);
+        int pos;
+
+        if(truck_ == true)
+        {
+            for(int count = 0; count < is_published_truck.size(); count++)
+                {
+                    if(is_published_truck[count]==false)//only check system alerts that have not been published
+                    {   
+                        pos = getStatusTruck(crit_driver_status_truck, time_now);
+                        if(count == pos)
+                           is_published_truck[count] = true;
+                    }
+                }
+         }//End truck
+        
+        else if (car_ == true)
+        {
+            for(int count = 0; count < is_published_car.size(); count++)
+                {
+                    if(is_published_car.at(count) == false)//only check system alerts that have not been published
+                    {   
+                        pos = getStatusCar(crit_driver_status_car, time_now);
+                        if(count == pos)
+                            is_published_car[count] = true;
+
+                    }
+                }
+
+        }//End car
+
+    }//End spinObject
+
+
+    int HealthMonitor::getStatusTruck(std::string status, long time_now)
+    {
+        ros::Duration sd(startup_duration_);
+        long start_duration=sd.toNSec() / 1e6;
+
+        if(status == "s_1_l1_1_l2_1_g_1")        
+            return 0;
+        else if(status == "s_1_l1_0_l2_1_g_1")
+            return 2;
+        else if (status == "s_1_l1_0_l2_1_g_0")
+            return 3;
+        else if(status == "s_1_l1_1_l2_1_g_0")
+            return 4;
+        else if(status =="s_1_l1_0_l2_0_g_1")
+            return 5;
+        else if(status == "s_1_l1_0_l2_0_g_0") 
+            return 6;
+        else if(status =="s_0")
+            return 7;
+
+
+
+    }
+
+
+    int HealthMonitor::getStatusCar(std::string status, long time_now)
+    {
+        ros::Duration sd(startup_duration_);
+        long start_duration=sd.toNSec() / 1e6;
+
+        if(status == "s_1_l_1_g_1" )
+            return 0;
+        else if(status == "s_1_l_1_g_0")
+            return 2;
+        else if(status == "s_1_l_0_g_1")
+            return 3;
+        else if(status == "s_1_l_0_g_0")
+            return 4;
+        else if(status == "s_0")
+            return 5;
+    }
+
+
+    std::vector<bool> HealthMonitor::getPubStatusCar()
+    {
+        std::vector <bool> cr_;
+        for(int i=0; i< is_published_car.size();i++)
+            cr_.push_back(is_published_car[i]);
+        return cr_;
+    }
+
+    std::vector<bool> HealthMonitor::getPubStatusTruck()
+    {
+        return is_published_truck;
+    }
+
+
+    void HealthMonitor::setDriverManager(DriverManager dm)
+    {
+        driver_manager_ = dm;
+    }
+
+    void HealthMonitor::setCarTrue()
+    {
+        car_ = true;
+    }
+
+    void HealthMonitor::setTruckTrue()
+    {
+        truck_ = true;
```

Comment:
Probably worth setting car_ = false whenever truck_ = true and vice versa to prevent a possible mismatch

Reply:
Also would want to log in that event since it likely indicates an error if one was set true when the other was already true. 
Throwing an exception would also be acceptable.